[Wadia Ganim] iP

wadia82 opened the pull request:
# BLOOP

>"Your mind is for having ideas, not holding them.” – David Allen

Bloop frees your mind of having to remember things you need to do. It's,
-text-based
-easy to learn
-~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2223S1/ip/pull/386).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 💯 

And it is **FREE**!

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

##
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public static void main(String[] args) {
        chat();
}
```

Commit:
Add Gradle support

wadia82 committed:
Called chatbot Bloop and added greet, echo and exit

wadia82 committed:
Added list feature

wadia82 committed:
Added feature to mark tasks as done or not

wadia82 committed:
Added different kinds of tasks

wadia82 committed:
automated testing

wadia82 committed:
Handle errors

wadia82 committed:
Add support for deleting tasks from the list

wadia82 committed:
no message

wadia82 committed:
Save tasks in hard disk

wadia82 committed:
Add support for understanding date and time

wadia82 committed:
Deal with merge conflicts

wadia82 committed:
Add classes to improve OOP

wadia82 committed:
Add classes to package

wadia82 committed:
Merge branch 'add-gradle-support'

wadia82 committed:
Automate project builds using gradle

wadia82 committed:
Add JUnit tests to test the behavior of the code

wadia82 committed:
Add JavaDoc

wadia82 committed:
Tweak code to comply with coding standard

wadia82 committed:
Add support for finding a task by searching for a keyword

wadia82 committed:
no message

wadia82 committed:
Merge branch 'branch-A-CodingStandard'

wadia82 committed:
Merge branch 'branch-Level-9'

wadia82 committed:
Specify format for date and time input

wadia82 committed:
Edit exception message

tsammeow commented on src/main/java/bloop/Task.java:
```diff
@@ -0,0 +1,64 @@
+     * @param dateTime LocalDateTime object with the date and time entered by the user.
+     * @return Formatted date and time.
+     */
+    public String formatDateTime(LocalDateTime dateTime) {
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("dd MMM yyyy hh:mm a");
+        return dateTime.format(outputFormatter);
+    }
```
Perhaps this should be a `static` method?

tsammeow commented on src/main/java/bloop/TaskList.java:
```diff
@@ -0,0 +1,149 @@
+     */
+    public TaskList(Ui ui, Storage storage) {
+        this.ui = ui;
+        tasks = new ArrayList<>();
```
Did you intend to overwrite the static `tasks` field every time a new `TaskList` is created?

tsammeow commented on src/main/java/bloop/TaskList.java:
```diff
@@ -0,0 +1,149 @@
+public class TaskList {
+
+    /** list of tasks */
+    private static ArrayList<Task> tasks;
```
Perhaps the list of tasks should be an instance field instead of a static one?

tsammeow commented on src/main/java/bloop/Ui.java:
```diff
@@ -0,0 +1,46 @@
+ * Interactions with the user.
+ */
+public class Ui {
+    private static final String HI_MESSAGE = "Hey! I'm bloop.Bloop\n" + "\tWhat can I do for you?";
+
+    private static final String BYE_MESSAGE = "Goodbye! Hope to see you soon :)";
```
Should the names of these constants be modified to have a common prefix?

tsammeow reviewed (changes requested):
Your code was overall easy to read and it was clear what everything did. Apart from some minor issues, everything looks good!

wadia82 committed:
Add and use checkstyle

gerardstevan commented on src/main/java/bloop/Parser.java:
```diff
@@ -0,0 +1,64 @@
+     * @throws BloopException If the command is not recognised or incomplete.
+     * @throws IOException If there is a problem writing to file.
+     */
+    public void parse(String text) throws BloopException, IOException {
```
According to the code-quality guide, you should do something if a function is more than 30 LOC.

gerardstevan reviewed (commented):
Overall, I like how your code follows the code quality guidelines and is easy to read.

wadia82 committed:
Add GUI

wadia82 committed:
Improve code quality

wadia82 committed:
Add assertions

wadia82 committed:
Revert "Add assertions"

This reverts commit 983ac0b7b9cf9220a49ce055b9ff7d388a487349.

wadia82 committed:
Add assertions

Might make a mistake while defining and calling functions.

The mistakes might lead to the program not functioning as expected.

Asserting the function is defined correctly.

wadia82 committed:
Merge pull request #1 from wadia82/branch-A-Assertions

Branch-A-Assertions

wadia82 committed:
Merge branch 'master' of https://github.com/wadia82/ip

wadia82 committed:
Merge branch 'branch-A-CodeQuality'

wadia82 committed:
Add support to sort deadlines by date and time

Deadlines are not sorted.

User should be able to know the closest deadline.

Let's sort and list them out in order of when they are due.

wadia82 committed:
Add user guide

wadia82 committed:
Rename Ui

wadia82 committed:
Correct code style and create jar file
